### ISSUE TITLE
Problem with loss function with 512x512 data on motion cosegmentation model

### ISSUE THREAD

**Issue by playmakerbugger** [adeptflax/motion-models#4]
I am training the motion cosegmentation model on a 512 x 512 set of data, using 4 gpus. I am having a stall during the loss process where it locks during the calculation of the loss. It gives a user warning of "was asked to gather along dimension 0, but all input tensors were scalars; will instead unsqueeze and return a vector."

I was wondering if you encountered this problem when you were retraining with high res data. I noticed that the loss is only calculating on gpu 0.
